examples/dump-decls/CMakeLists.txt

```diff
@@ -1,2 +1,8 @@
+find_package(Boost REQUIRED COMPONENTS iostreams)
```

Comment:
I'm not sure why `dump-decls` should depend on Boost::iostreams and explicitly include `Boost_INCLUDE_DIR`. Maybe it was true before the commit 4d50caf8a8d7c9b2955f3a13afc727fd53e595b5, but I'd prefer to leave `CMakeLists.txt` of `dump-decls` as simple as possible, it should be literally [two-line](https://github.com/AndreyG/ifc-reader/blob/master/examples/dump-decls/CMakeLists.txt).

Reply:
At least whole project did not compile without this bugfix at least on my machine. Does it compiles for you out of box ?

Reply:
Yes, of course, it compiles for me. And it compiles on [CI](https://github.com/AndreyG/ifc-reader/actions).
Could you please try compile the latest version from master, I have removed boost::iostream from public headers of `ifc-core`.

Reply:
I've originally had quite a lot of pain of bringing required libraries - but now noticed that for linux this works so simply by just apt-get:ing required packages. Nice. Wish this would work as smoothly for windows.

But yes, haven't noticed that you had CI enabled, otherwise would try to activate windows support as well.
Suspect for windows you would need to run cmake configure / cmake build 2 times.

But with vcpkg I've got some more or less random errors , related to vcpkg itself.

Wondering if you can try to active cmake + vcpkg bootstrapper on linux as well. There should be bootstrap-vcpkg.sh, otherwise commands should work identically.

Using vcpkg is bit hacky approach, maybe better approach would be to use boost iostreams git directly.

But vcpkg also patches some of libraries if original git misses them, which in a turn is better from windows compilation perspective.

Reply:
Actually, I'm developing on Windows using vcpkg. I have installed required dependencies, like this
```
vcpkg install boost-iostreams:x64-windows nlohmann-json:x64-windows
```
and then just use `CMAKE_TOOLCHAIN_FILE` (see the last paragraph of https://github.com/microsoft/vcpkg#quick-start-windows):
```
cmake -G "Visual Studio 16 2019" -A x64 -DCMAKE_TOOLCHAIN_FILE=[path to vcpkg]/scripts/buildsystems/vcpkg.cmake
```